packages/wallet-manager/src/WalletManagerController.ts

```diff
@@ -0,0 +1,155 @@
+import { ReactiveControllerHost } from 'lit';
+import { EvmWallet, SubstrateWallet } from '.';
+import { Web3Provider } from '@ethersproject/providers';
+import { ApiPromise } from '@polkadot/api';
+import { AddChain } from './types';
+import { customEVMEvents, IWalletManagerController } from './interfaces';
+import { Signer } from '@ethersproject/abstract-signer';
+
+export class WalletManagerController implements IWalletManagerController {
+  private host: ReactiveControllerHost;
+  evmWallet?: EvmWallet;
+  substrateWallet?: SubstrateWallet;
+  account?: string;
+  substrateAccount?: string;
+
+  constructor(
+    host: ReactiveControllerHost,
+    initArguments?: {
+      web3Provider?: Web3Provider;
+      apiPromise?: ApiPromise;
+      wssConnectionUrl?: string;
+    }
+  ) {
+    (this.host = host).addController(this);
+
+    if (initArguments?.web3Provider) {
+      this.initWeb3Provider(initArguments.web3Provider);
+    } else if (initArguments?.apiPromise) {
+      this.connectFromApiPromise(initArguments.apiPromise);
+    } else if (initArguments?.wssConnectionUrl) {
+      this.connectFromWssProvider(initArguments.wssConnectionUrl);
+    } else {
+      this.initWeb3Provider();
+    }
+  }
+
+  hostDisconnected(): void {
+    if (this.evmWallet) {
+      this.evmWallet?.removeListener(customEVMEvents.ACCOUNT_CHANGE, () => {});
+      this.evmWallet?.removeListener(customEVMEvents.CHAIN_CHANGE, () => {});
```

Comment:
these won't remove added listeners, since those are not listener callbacks. maybe it'd be better to use `removeAllListeners` for the event or have callbacks as private methods and remove them precisely.